conans/client/downloaders/caching_file_downloader.py

```diff
@@ -61,6 +61,7 @@ def _caching_download(self, urls, file_path,
             remove_if_dirty(cached_path)
 
             if os.path.exists(cached_path):
+                self._file_downloader.check_checksum(cached_path, md5, sha1, sha256)
```

Comment:
This won't cover all cases of ``.dirty``, only controlled ones. There can be some cases where things will crash and still the ``.dirty`` file will be there.

Reply:
Yup! That's why I was not confident in pushing this without being a draft, I still need to check those cases :)

Thanks for the heads up!